packages/block-editor/src/components/block-edit/edit.js

```diff
@@ -48,27 +57,225 @@ const Edit = ( props ) => {
 const EditWithFilters = withFilters( 'editor.BlockEdit' )( Edit );
 
 const EditWithGeneratedProps = ( props ) => {
-	const { attributes = {}, name } = props;
+	const { name, clientId, attributes, setAttributes } = props;
+	const registry = useRegistry();
 	const blockType = getBlockType( name );
 	const blockContext = useContext( BlockContext );
+	const registeredSources = useSelect( ( select ) =>
+		unlock( select( blocksStore ) ).getAllBlockBindingsSources()
+	);
+
+	const blockBindings = useMemo(
+		() =>
+			replacePatternOverridesDefaultBinding(
+				name,
+				attributes?.metadata?.bindings
+			),
+		[ attributes?.metadata?.bindings, name ]
+	);
```

Comment:
Can't this be moved within `useSelect`?

Reply:
I think it's used also by useCallback, but it could be moved within both maybe.

Reply:
In https://github.com/WordPress/gutenberg/pull/67523/commits/f192c04c50a6f6764ef978c2085ca2f307be2e51, I merged it with another `useMemo` instead.